src/components/ActionBar/ActionBarHeader.tsx

```diff
@@ -62,15 +58,13 @@ export const ActionBarHeader = forwardRef<HTMLHeadingElement, ActionBarHeaderPro
               </div>
               <div className={classes.actionBarTexts}>
                 <Paragraph
-                  as={renderAsElem}
```

Comment:
Yeah, i'll just do the same as designsystem and switch to using Heading component instead of Paragraph component which I think is more correct dom-wise, accessibility-wise and component-wise (since I'd expect Paragraph to have nothing to do with html-header elements but with p-element).

Reply:
Sounds good 👍
Just please make sure the size matches the current one. The reason why we used the Paragraph component in the first place is because the Heading component in the design system was too large on all levels. The use of the as-prop has till now made sure the dom and accessibility concerns were covered, since it rendered the Paragraph as an h-element instead of a p-element while still keeping the styling of the Paragraph component.

However, I think they might have made smaller headings available now so maybe xsmall or xxsmall (if available) can be used (you'll have to try and compare with how it currently looks in AT)

That being said, I would still encourage you to take a look at the implementation of the Accordion component in the design system, where they encapsulate the entire button inside a Heading component and then use Paragraph to write the actual text. I'm not sure whether that's a better alternative than having a button with an h-element inside of it, but it's definitely an option.

Reply:
I've worked the whole day with a bug in the design system to try to make asChild prop work and it turns out it doesn't work as I would have expected it to work. When I look now at the implementation in FDS they have a header-element outside Paragraph component yeah, but they also have a <p> element as child of Paragraph, I think this is is sub-optimal. I just needed to write this down to remember it after holiday.

Should we do something about this, what you think? I think they need to fix their asChild-prop at some point to work exactly like in Radix because now it's kind of like a fake asChild and you need to add a redundant element to the dom as well when using it.

Reply:
If design system doesn't have plans to fix the asChild prop I'd make an argument that it's better to set headerValue in the title than by us having a headingLevel-prop as that also removes one layer in the dom.

Reply:
Like their asChild prop works now and with us using the Heading-component the dom will be :
<h5>
   <p>
     {title}
   </p>
</h5>

but if we skip using their Heading component the dom will be:

<h5>
  {title}
</h5>

it also reduces our code complexity as we can remove one prop and the code that sets headingLevel.

Reply:
After thinking about this and seeing how designsystemet and Radix UI does it I'll change one of the outer divs in the ActionBar to become a header instead